### ISSUE TITLE
KeePassXC-Browser does not connect to browsers using flatpak kepassxc 2.4.6 on Deepin V20.2

### ISSUE THREAD

**Issue by serial02** [keepassxreboot/keepassxc#6373]
KeePassXC-Browser does not connect to browsers using flatpak kepassxc 2.4.6 on Deepin V20.2

All KeePassXC-Browser extensions in browsers are updated to the latest version, being version 1.7.7.

![Capturar_select-area_20210406121942](https://user-images.githubusercontent.com/32621790/113735277-7aa71180-96d2-11eb-94dd-d0517667543b.png)

Opera, Brave, Deepin Browser do not connect.

![Deepin navegador](https://user-images.githubusercontent.com/32621790/113735562-bb068f80-96d2-11eb-85a8-1853e64c45b8.gif)

Brave
https://imgur.com/IkJjhGa

Opera
https://imgur.com/BAUndYJ

Microsoft Edge browser can connect using the following configuration
and firefox connects normally.

/home/USER/.config/microsoft-edge-dev/NativeMessagingHosts

Configuration:

OS: Deepin 20.2 x86_64                               
Host: p6540br xxx0204GRxxxxxxxx0                     
Kernel: 5.11.1-amd64-desktop                         
Uptime: 1 hour, 24 mins                              
Packages: 1709 (dpkg), 16 (flatpak)                  
Shell: bash 5.0.3                                    
Resolution: 1920x1080                                
DE: Deepin                                           
WM: KWin                                             
Icons: bloom [GTK2/3]                                
Terminal: deepin-terminal                            
CPU: Intel i3 540 (4) @ 3.067GHz                     
GPU: Intel Core Processor                            
Memory: 2719MiB / 7695MiB

**Comment by droidmonkey** [comment]
We don't support opera or deepin. Brave may require a custom browser setting.

**Comment by serial02** [comment]
> We don't support opera or deepin. Brave may require a custom browser setting.

Previously everything worked normally, after reinstalling deepin V20.2 on 05/04/2021 it did not accept the new settings.

**Comment by droidmonkey** [comment]
Why do you suspect this is a keepassxc issue than?

**Comment by serial02** [comment]
> Why do you suspect this is a keepassxc issue than? I am closing this since we do not support the browser in question.

The browsers in question without exception are the Chromium base, where after updating the extension that these problems began to occur. Everything leads to believe that it is the browser extension that does not connect with keepassxc 2.6.4.

**Comment by serial02** [comment]
> Why do you suspect this is a keepassxc issue than? I am closing this since we do not support the browser in question.

Hello everyone

I would like help to understand why Fedora 34 Beta flavor Deepin V20 works correctly and connect to Opera Browser normally and this same application and extension does not work correctly on Deepin V20.2. Can anyone have an explanation for this? Could it be because Keepassxc 2.6.4 is installed on Deepin V20.2 via flatpak?

![DeepinScreenshot_20210408105212](https://user-images.githubusercontent.com/32621790/114054808-8e837c80-9866-11eb-8a5b-20a597688026.png)

![DeepinScreenshot_dde-desktop_20210408123347](https://user-images.githubusercontent.com/32621790/114054968-b4a91c80-9866-11eb-9ba5-ae8faca2d559.png)

**Comment by droidmonkey** [comment]
Use the appimage instead of flatpak to confirm your question.

**Comment by serial02** [comment]
> Use the appimage instead of flatpak to confirm your question.

Thank You

**Comment by serial02** [comment]
> Use the appimage instead of flatpak to confirm your question.

@droidmonkey 

The keepassxc appimage works only the Brave Browser. The Opera and Deepin browsers do not work using appimage.

Here are the screenshots:

Brave connects OK using appimage on Distro Deepin V20.2 

![Capturar_dde-desktop_20210408135121](https://user-images.githubusercontent.com/32621790/114066570-3d798580-9872-11eb-8679-ca68f0e903df.png)

Opera does not connect using appimage and flatpak on Distro Deepin V20.2 

![Capturar_dde-desktop_20210408134910](https://user-images.githubusercontent.com/32621790/114066951-a3660d00-9872-11eb-9b65-acd1f8377da2.png)

Browser Deepin Opera does not connect using appimage and flatpak on Distro Deepin V20.2 

![Capturar_org deepin browser_20210408135414](https://user-images.githubusercontent.com/32621790/114067048-bf69ae80-9872-11eb-9fdd-965fe11b8934.png)

**Comment by droidmonkey** [comment]
Yes as I stated previously we don't support Opera and Deepin browsers. You'll have to use the custom browser option in the advanced browser integration settings.

**Comment by serial02** [comment]
> Yes as I stated previously we don't support Opera and Deepin browsers. You'll have to use the custom browser option in the advanced browser integration settings.

Only if you make room to make more custom settings, because there is already one and apparently only accepts one, if I'm wrong. If you have another suggestion and can teach me how to do it, I will be very grateful!

![Capturar_keepassxc_20210408220327](https://user-images.githubusercontent.com/32621790/114114000-4429ec00-98b6-11eb-9c39-70e5f6bec363.png)

**Comment by droidmonkey** [comment]
We only support one custom browser.

**Comment by serial02** [comment]
> We only support one custom browser in the application, but you can manually copy the native messaging json to the other browsers config folder and everything should work fine.


I managed to do what you suggested and all browsers are working perfectly with Keepassxc 2.6.4 installed Via Flatpak on Deepin V20.2 with the KeePassXC-Browser - 1.7.7 extension.

Follows images of the browsers showing that everything is ok. Thank you! 

Deepin Browser

![Capturar_20210409120133](https://user-images.githubusercontent.com/32621790/114201358-7c6c1180-992c-11eb-9c13-dc53bfa85626.png)

Brave Browser

![Capturar_20210409120154](https://user-images.githubusercontent.com/32621790/114201458-960d5900-992c-11eb-82a0-5be8f11b2029.png)

Opera Browser

![Capturar_20210409120235](https://user-images.githubusercontent.com/32621790/114201613-c5bc6100-992c-11eb-8c0e-62fbbcc55a70.png)

**Comment by droidmonkey** [comment]
Welcome

**Comment by serial02** [comment]
> Welcome

Hello

After updating KeePassXC-Browser - 1.7.8.1 all browsers no longer connect to Keepassxc flatpak 2.6.4. I went to see the settings and they are the ones that were previously, following your suggestion and that were working normally. 

Clicking reload on the browser extension opens the keepassxc 2.6.4 application instead of connecting.

![Capturar_20210414140525](https://user-images.githubusercontent.com/32621790/114750712-a8bbcf80-9d2a-11eb-838e-662652740adf.png)


What can it be this time? Any suggestion?

![Capturar_select-area_20210414140156](https://user-images.githubusercontent.com/32621790/114750268-2e8b4b00-9d2a-11eb-9c67-44f6c76cbe3e.png)

**Comment by droidmonkey** [comment]
Don't use the flatpak, we don't currently support that distribution. Use the appimage instead.

**Comment by serial02** [comment]
> Don't use the flatpak, we don't currently support that distribution. Use the appimage instead. Not sure if you tried restarting your browser as well.

Yes, I restarted the browsers, the PC and tried the many tips on the page here on the forum, but I was successful. I also tried appimage and was unsuccessful. The only thing I was able to do was leave an exit showing what happened. 

I hope it helps and soon keepassxc flatpak will work again.

![Capturar_select-area_20210415142217](https://user-images.githubusercontent.com/32621790/114918626-a7a7a280-9dfd-11eb-9f17-e0debdcee5fe.png)
